src/query.c

```diff
@@ -833,10 +833,8 @@ IndexIterator *QAST_Iterate(const QueryAST *qast, const RSSearchOptions *opts, R
   };
   IndexIterator *root = Query_EvalNode(&qectx, qast->root);
   if (!root) {
-    if (!QueryError_HasError(status)) {
-      // Return the dummy iterator
-      return NewEmptyIterator();
-    }
+    // Return the dummy iterator
```

Comment:
why? you're just suppressing error reporting.

Reply:
The error is not passing to anyone here so there is not way something set it, this was a dead code.